tezos/context/src/tezedge_context.rs

```diff
@@ -70,98 +72,125 @@ impl TezedgeIndex {
+    pub fn get_hash(&self, hash_id: HashId) -> Result<Option<ObjectHash>, DBError> {
         Ok(self
             .repository
             .read()?
             .get_hash(hash_id)?
             .map(|h| h.into_owned()))
     }
 
-    pub fn get_entry(&self, hash: HashId, storage: &mut Storage) -> Result<Entry, MerkleError> {
-        match self.find_entry(hash, storage)? {
-            None => Err(MerkleError::EntryNotFound { hash_id: hash }),
-            Some(entry) => Ok(entry),
+    /// Fetches data from the repository and deserialize it into `storage`.
+    ///
+    /// Returns an error when the object was not found.
+    pub fn get_object(&self, hash: HashId, storage: &mut Storage) -> Result<Object, MerkleError> {
+        match self.find_object(hash, storage)? {
+            None => Err(MerkleError::ObjectNotFound { hash_id: hash }),
+            Some(object) => Ok(object),
         }
     }
 
+    /// Fetches the commit associated to this `hash` from the repository.
     pub fn find_commit(
         &self,
         hash: HashId,
         storage: &mut Storage,
     ) -> Result<Option<Commit>, DBError> {
-        match self.find_entry(hash, storage)? {
-            Some(Entry::Commit(commit)) => Ok(Some(*commit)),
-            Some(Entry::Tree(_)) => Err(DBError::FoundUnexpectedStructure {
+        match self.find_object(hash, storage)? {
+            Some(Object::Commit(commit)) => Ok(Some(*commit)),
+            Some(Object::Directory(_)) => Err(DBError::FoundUnexpectedStructure {
                 sought: "commit".to_string(),
                 found: "tree".to_string(),
             }),
-            Some(Entry::Blob(_)) => Err(DBError::FoundUnexpectedStructure {
+            Some(Object::Blob(_)) => Err(DBError::FoundUnexpectedStructure {
                 sought: "commit".to_string(),
                 found: "blob".to_string(),
             }),
             None => Ok(None),
         }
     }
 
+    /// Fetches the commit associated to this `hash` from the repository.
+    ///
+    /// Returns an error when the commit was not found.
     pub fn get_commit(&self, hash: HashId, storage: &mut Storage) -> Result<Commit, MerkleError> {
         match self.find_commit(hash, storage)? {
-            None => Err(MerkleError::EntryNotFound { hash_id: hash }),
-            Some(entry) => Ok(entry),
+            None => Err(MerkleError::ObjectNotFound { hash_id: hash }),
+            Some(object) => Ok(object),
         }
     }
 
-    pub fn find_tree(&self, hash: HashId, storage: &mut Storage) -> Result<Option<Tree>, DBError> {
-        match self.find_entry(hash, storage)? {
-            Some(Entry::Tree(tree)) => Ok(Some(tree)),
-            Some(Entry::Blob(_)) => Err(DBError::FoundUnexpectedStructure {
+    /// Fetches the directory associated to this `hash` from the repository.
+    pub fn fetch_directory(
+        &self,
+        hash: HashId,
+        storage: &mut Storage,
+    ) -> Result<Option<DirectoryId>, DBError> {
```

Comment:
The naming we have is not very consistent, we have at least `fetch_`, `find_` and `get_`, and with different semantics each. I would like to have `find_` for anything that will return an `Option`, `fetch_` for versions that don't return an option and will return an error, and not sure about `get_`, I think the other two should cover it.

Reply:
Commits for example have find + get, and here directory has fetch + get.